server/src/app/app.ts

```diff
@@ -87,7 +95,11 @@ app.post("/bots", async (req, res) => {
     bots = worker.generateNewBots(reqBots);
     generation = req.body.generation + 1;
   }
-  MetadataCache.addGeneration(bots, generation);
+
+  const timestamps = {startTime : req.body.startTime, endTime : req.body.endTime}
```

Comment:
I think it's easier to generate this information on the backend then have it generated in the frontend then send it to the backend.
Right now, we already keep track of `startTime` (see `newSession`), and it makes more sense to track it on a session basis then a request basis (since it stays constant for a given session).
Adding a `endTime` stamp is also fairly trivial.

Reply:
Hm.. I thought about this as well.
I personally figured it would be cleaner to do it on the front end rather than solely in the backend because this means a session gets written once and then never again.
For us to do timestamps on the backend requires us to:
````
1. set the startTime when the session begins in the first generation
2. fetch that startTime and pair it with the endTime when the endpoint gets hit again (because `addGeneration` would otherwise just push)
3. Use the endTime to be the start time of the next new session 
4. Mutate that new session the next time the endpoint gets hit
5. And onwards...
````

Felt like a lot of state management :P .